Spock-core/src/Web/Spock/Internal/Wire.hs

```diff
@@ -332,51 +331,58 @@ applyAction req env (selectedAction : xs) =
 
 handleRequest
     :: MonadIO m
-    => SpockMethod
-    -> Maybe Word64
+    => SpockConfigInternal
+    -> SpockMethod
     -> (forall a. m a -> IO a)
     -> [ActionT m ()]
     -> InternalState
     -> Wai.Application -> Wai.Application
-handleRequest stdMethod mLimit registryLift allActions st coreApp req respond =
+handleRequest config stdMethod registryLift allActions st coreApp req respond =
     do reqGo <-
-           case mLimit of
+           case sci_maxRequestSize config of
              Nothing -> return req
              Just lim -> requestSizeCheck lim req
-       handleRequest' stdMethod registryLift allActions st coreApp reqGo respond
+       handleRequest' config stdMethod registryLift allActions st coreApp reqGo respond
 
 handleRequest' ::
     MonadIO m
-    => SpockMethod
+    => SpockConfigInternal
+    -> SpockMethod
     -> (forall a. m a -> IO a)
     -> [ActionT m ()]
     -> InternalState
     -> Wai.Application -> Wai.Application
-handleRequest' stdMethod registryLift allActions st coreApp req respond =
+handleRequest' config stdMethod registryLift allActions st coreApp req respond =
     do actEnv <-
            (Left <$> makeActionEnvironment st stdMethod req)
            `catch` \(_ :: SizeException) ->
-               return (Right sizeError)
+               return (Right $ getErrorHandler config status413)
        case actEnv of
          Left (mkEnv, vaultVar, cleanUp) ->
              do mRespState <-
-                    registryLift (applyAction req mkEnv allActions)
-                    `catch` \(_ :: SizeException) ->
-                        return (Just sizeError)
-                    `catch` \(e :: SomeException) ->
+                    registryLift (applyAction config req mkEnv allActions) `catches`
+                      [ Handler $ \(_ :: SizeException) ->
+                          return (Just $ getErrorHandler config status413)
+                      , Handler $ \(e :: SomeException) ->
                         do putStrLn $ "Spock Error while handling " ++ show (Wai.pathInfo req) ++ ": " ++ show e
-                           return $ Just serverError
+                           return $ Just $ getErrorHandler config status500
+                      ]
                 cleanUp
                 case mRespState of
-                  Just respState ->
+                  Just respState @ (ResponseState _ _ _ _) ->
```

Comment:
You could just flip this pattern match with the next pattern match so that you don't need to mach with all those wildcards. Alternatively you could use `ResponseState {}`